omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangePasswordPageController.java

```diff
@@ -110,12 +110,13 @@ public String post(PageModel model, @SpringBean("userService") UserService userS
 				} else {
 					nextPage = "myaccount/myAccount";
 				}
+
+				Context.refreshAuthenticatedUser();
+				request.getSession().invalidate();
```

Comment:
Does this forcibly log the user out?

Reply:
> Does this forcibly log the user out?

Yes @ibacher  The user is forcefully logged out.  Thank you

Reply:
maybe i should remove line 115 and add the code below to logout the user using the context so that he is redirected to the login screen.  What is the better option? 
```
 Context.logout();
 request.getSession().invalidate();
request.getSession().setAttribute("manual-logout", "true");
```

Reply:
I mean, the real ideal would be if there's a way for a user to change their password and have the session continue uninterrupted, but, yeah, if we're going to log the user out, maybe just do the whole thing. We could even add a redirect to the login page here?